internal/rest/resources/control.go

```diff
@@ -84,21 +84,21 @@ func controlPost(state state.State, r *http.Request) response.Response {
 	reverter := revert.New()
 	defer reverter.Fail()
 	reverter.Add(func() {
-		reExec, err := resetClusterMember(r.Context(), state, true)
+		// Run the pre-remove hook like we do for cluster node removals.
+		err := intState.Hooks.PreRemove(r.Context(), state, true)
 		if err != nil {
+			logger.Error("Failed to run pre-remove hook on bootstrap error", logger.Ctx{"error": err})
 			return
```

Comment:
@masnax Should an error in the hook cancel the revert or should we also remove the `return` statement? I would vote for removing it.

Reply:
my 2ct: If something fails we need to call `resetClusterMember` otherwise a follow-up bootstrap try will fail.

Cleanups should always be on a best-effort basis. If something fails, move on to the next cleanup step.